otter/convergence/service.py

```diff
@@ -0,0 +1,58 @@
+"""Converger service"""
+
+from effect import Effect
+from effect.twisted import perform
+
+from twisted.application.service import Service
+
+from otter.constants import CONVERGENCE_LOCK_PATH
+from otter.convergence.composition import (
+    execute_convergence, get_desired_group_state)
+from otter.http import TenantScope
+from otter.util.deferredutils import with_lock
+
+
+class Converger(Service, object):
+    """Converger service"""
+
+    def __init__(self, reactor, kz_client, dispatcher):
+        self._reactor = reactor
+        self._kz_client = kz_client
+        self._dispatcher = dispatcher
+
+    def _get_lock(self, group_id):
+        """Get a ZooKeeper-backed lock for converging the group."""
+        path = CONVERGENCE_LOCK_PATH.format(group_id=group_id)
+        return self._kz_client.Lock(path)
```

Comment:
I am sorry you might have to setup acquire timeout before returning zk lock. This is different from `acquire_timeout` called below in `with_lock`. See https://github.com/rackerlabs/otter/blob/master/otter/models/cass.py#L754